.github/workflows/unit_testing_wf.yaml

```
@@ -0,0 +1,23 @@
+name: macrophage-striation-unittests
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  unit-test-core:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - uses: actions/checkout@v2
```

Consider using `checkout@v3` which is an updated version of this action.

```suggestion
      - uses: actions/checkout@v3
```